### ISSUE TITLE
Loss computation; SmoothL1 and BCE for coordinates

### ISSUE THREAD

**Issue by nirbenz** [TencentYoutuResearch/ObjectDetection-OneStageDet#8]
Thanks for this repo! I'm learning the loss function since it is quite different from other repos I've seen. Something specific that caught my attention is the loss computation for the coordinates:

https://github.com/TencentYoutuResearch/ObjectDetection-OneStageDet/blob/5f52c97e14378d4de25f59d4aaee7ffe43f3523d/yolo/vedanet/loss/_yololoss.py#L124-L125

So, *w/h* use the *SmoothL1* loss while the *x/y* coordinates use BCE. None of which are mentioned in the paper. Specifically, some detectors (namely Faster R-CNN and its variants) use SmoothL1 for all coordinates - but using BCE for *x/y* (regression targets) makes little sense to me. Can you please elaborate on this?

Thanks.

**Comment by nirbenz** [comment]
Indeed, smooth L1 _could_ achieve steadier gradient signal in the beginning of training and afterwards behave the same as L2 (or identically even). But using BCE for x/y is stranger to me. Why not use the same loss for all 4 coordinates?

**Comment by nirbenz** [comment]
Sorry about reopening - but what about using BCE for the center coordinates? Any intuition behind that?

**Comment by dlyldxwl** [comment]
@nirbenz @mileistone 
hi, I don't know the effect of coord_mask (2-S).
` coord_mask[b][best_n][0][gj*nW+gi] = 2 - anno.width*anno.height/(nW*nH*self.reduction*self.reduction)`
could you help me?
